src/WorldIDIdentityManagerImplV1.sol

```diff
@@ -212,30 +216,35 @@ contract WorldIDIdentityManagerImplV1 is
     ///      initialisations allowed, so decide carefully when to use them. Many cases can safely be
     ///      replaced by use of setters.
     ///
+    /// @param _treeDepth The depth of the MerkeTree
     /// @param initialRoot The initial value for the `latestRoot` in the contract. When deploying
     ///        this should be set to the root of the empty tree.
     /// @param _merkleTreeVerifier The initial tree verifier to use.
+    /// @param _semaphoreVerifier The initial semaphore verifier to use.
     /// @param _enableStateBridge Whether or not the state bridge should be enabled when
     ///        initialising the identity manager.
     /// @param initialStateBridgeProxyAddress The initial state bridge proxy address to use.
     ///
     /// @custom:reverts string If called more than once at the same initalisation number.
     function initialize(
+        uint8 _treeDepth,
         uint256 initialRoot,
         ITreeVerifier _merkleTreeVerifier,
+        ISemaphoreVerifier _semaphoreVerifier,
         bool _enableStateBridge,
         address initialStateBridgeProxyAddress
     ) public reinitializer(1) {
         // First, ensure that all of the parent contracts are initialised.
         __delegateInit();
 
         // Now perform the init logic for this contract.
+        treeDepth = _treeDepth;
```

Comment:
Please validate that this does not exceed the maximum depth allowed by Semaphore. It should revert with a descriptive error if so.

Reply:
Should we enable all depths supported by Semaphore or limit it to those supported in signup-sequencer (16, 20, 30)?